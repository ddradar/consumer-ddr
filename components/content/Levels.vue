<template>
  <NuxtLink v-for="(l, i) in slugs" :key="l" :to="l">
    <Badge>{{ levels[i] ?? levels[0] }}</Badge>
  </NuxtLink>
</template>

<script lang="ts" setup>
interface LevelProps {
  links: string
  level: string
}

const _props = defineProps<LevelProps>()
const slugs = _props.links.split(',')
const levels = _props.level.split(',')
</script>
