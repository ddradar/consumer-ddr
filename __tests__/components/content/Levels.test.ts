import { mount, RouterLinkStub } from '@vue/test-utils'
import { describe, expect, test } from 'vitest'

import Levels from '~~/components/content/Levels.vue'

describe('components/content/Levels.vue', () => {
  test.each([
    ['/playstation-jp/1st', '1'],
    ['/playstation-jp/1st,/playstation-jp/2nd', '2'],
    ['/playstation-jp/1st,/playstation-jp/2nd', '2,3'],
  ])('(links: %s, level: %s) renders expected badge', (links, level) => {
    const wrapper = mount(Levels, {
      global: { stubs: { Badge: true, NuxtLink: RouterLinkStub } },
      props: { links, level },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
