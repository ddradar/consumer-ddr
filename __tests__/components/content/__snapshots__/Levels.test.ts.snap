// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/content/Levels.vue > (links: /playstation-jp/1st, level: 1) renders expected badge 1`] = `
<div
  data-v-app=""
>
  
  <a>
    <badge-stub />
  </a>
  
</div>
`;

exports[`components/content/Levels.vue > (links: /playstation-jp/1st,/playstation-jp/2nd, level: 2) renders expected badge 1`] = `
<div
  data-v-app=""
>
  
  <a>
    <badge-stub />
  </a>
  <a>
    <badge-stub />
  </a>
  
</div>
`;

exports[`components/content/Levels.vue > (links: /playstation-jp/1st,/playstation-jp/2nd, level: 2,3) renders expected badge 1`] = `
<div
  data-v-app=""
>
  
  <a>
    <badge-stub />
  </a>
  <a>
    <badge-stub />
  </a>
  
</div>
`;
